### ISSUE TITLE
diff path emptied

### ISSUE THREAD

**Issue by Berkmann18** [angus-c/just#121]
I'm using both `just-diff` and `just-diff-apply` and I noticed that once an array of diffs (I called it `dif`) is passed into `diffApply`, the `path` field of the objects in that array are empty which I didn't expect and I'm not sure I should have expected that.

Example code:
```js
const { diff } = require('just-diff')
const { diffApply } = require('just-diff-apply')

let old = [['Hello', 'world']]
let cur = [['Hello', 'all']]

let dif = diff(old, cur) //[{ op: 'replace', path: ['0', '1'], value: 'all }]
let copy = [...old]
let fixed = diffApply(copy, dif) // ['Hello', 'all']
copy; // [['Hello', 'all']]
dif; //[{ op: 'replace', path: [], value: 'all' }]
```

**Comment by angus-c** [comment]
Do you mean you're surprised `dif` appears to get mutated by `diffApply`? That shouldn't happen, I'll take a look.

**Comment by angus-c** [comment]
Fixed in `just-diff-apply@2.0.0`

```js
const { diff } = require('just-diff')
const { diffApply } = require('just-diff-apply')

let old = [['Hello', 'world']]
let cur = [['Hello', 'all']]

let dif = diff(old, cur) //[{ op: 'replace', path: ['0', '1'], value: 'all }]
let copy = [...old]
let fixed = diffApply(copy, dif) // ['Hello', 'all']
copy; // [['Hello', 'all']]
dif; //[{ op: 'replace', path: ['0', '1'], value: 'all' }]
```

**Comment by Berkmann18** [comment]
@angus-c 
> Do you mean you're surprised `dif` appears to get mutated by `diffApply`? That shouldn't happen, I'll take a look.

Yes.
